### ISSUE TITLE
How to configure Pact File generation folder

### ISSUE THREAD

**Issue by kmh0237** [SEEK-Jobs/pact-net#42]
Hello, I am working on generating a pact file at the moment, which works fine when I run my tests in my project from Visual Studio.  However, when I run those same tests from the command line as an administrator, no pact file is generated.  I was hoping this could work from the command line (currently using MsTest.exe) so the build process could generate a pact.  As a side note, the pact file would normally generate under the folder Api.Tests, which I assume is because it is the root folder of my test project.  Any insight on this would be appreciated.

![cmd](https://cloud.githubusercontent.com/assets/986564/8356665/4d805672-1b1c-11e5-98cd-b011bc1fc47e.PNG)

Thanks!

**Comment by kmh0237** [comment]
Okay, after some more testing I found out that MsTest will generate a folder called TestResults in the folder you are running the command from (in my case above, VisualStudio\Common7\IDE\).  In this folder structure, it generates two subfolders and copies the dlls into them.  The pact files appear two directories back (..\..\) from where the dlls are.  I observed this behavior with VsTest as well, except instead of copying the dlls, it would just generate the pact folder two folders back from the dlls you are using to run the tests.

Anyways, the main question I still have is this: is there any way to configure the tests to generate the pacts folder in a different place?

**Comment by neilcampbell** [comment]
Hey @kmh0237. At this stage there is no way to configure this, however it is an idea I have been planning on adding, so will get that sorted for you.

**Comment by abhayachauhan** [comment]
+1

We have a similiar issue at the moment - I have spent a little time looking at pushing our generated pact files from our FAKE build process - and I'm finding the "pacts" directory is created by the files themselves are missing... Haven't investigated what is going on but something is disconnected...

I can assist with this if required. This should be a small change...

**Comment by neilcampbell** [comment]
@abhayachauhan I think that may be a separate thing, I have never seen that one happen. It's probably worth deleting the pacts dir and trying again, just to make sure its in a clean state. Definitely something strange, we create the dir (where required) immediately before creating the file, so its very odd if the dir can be created but the file cant and there are no exceptions. Out of interest what test framework are you using? Also can you see the logs dir and file?

**Comment by neilcampbell** [comment]
Hi @kmh0237
I have added support now for supplying a pact dir and will build a new version of the nuget package tomorrow.

Specifying a pact dir or log dir can be done using the following:

```C#
var config = new PactConfig { PactDir = @"..\pacts", LogDir = @"c:\temp\logs" }
var pactBuilder = new PactBuilder(config);
```

**Comment by neilcampbell** [comment]
New nuget package has now been published changes are included in 1.0.5

**Comment by neilcampbell** [comment]
@abhayachauhan Can you please create a new issue for the issue you are seeing?
